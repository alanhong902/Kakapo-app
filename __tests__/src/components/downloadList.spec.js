import React from 'react';
import renderer from 'react-test-renderer';
import { DownloadList } from 'components/';
import { getData, randomSounds } from '../helper';

test('<DownloadList/> render empty', () => {
  const props = { ...getData('sounds') };
  const tree = renderer.create(<DownloadList {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('<DownloadList/> render', () => {
  const props = { ...getData('sounds'), sounds: randomSounds(4) };
  const tree = renderer.create(<DownloadList {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('<DownloadList/> render sounds with `progress` < 1', () => {
  const props = { ...getData('sounds'), sounds: randomSounds(4) };
  const tree = renderer.create(<DownloadList {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
