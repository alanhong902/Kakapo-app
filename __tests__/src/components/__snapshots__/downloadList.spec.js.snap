// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DownloadList/> render 1`] = `
<div
  className="download-list"
>
  <div
    className="download active"
  >
    <div
      className="no-image"
    />
    <span
      className="title"
    >
      thunder
    </span>
    <div
      className="progress"
    >
      <div
        className="progress-barberpole"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <div
        className="progress-text"
      >
        50
        %
      </div>
    </div>
  </div>
  <div
    className="download active"
  >
    <div
      className="no-image"
    />
    <span
      className="title"
    >
      thunder
    </span>
    <div
      className="progress"
    >
      <div
        className="progress-barberpole"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <div
        className="progress-text"
      >
        50
        %
      </div>
    </div>
  </div>
  <div
    className="download active"
  >
    <div
      className="no-image"
    />
    <span
      className="title"
    >
      thunder
    </span>
    <div
      className="progress"
    >
      <div
        className="progress-barberpole"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <div
        className="progress-text"
      >
        50
        %
      </div>
    </div>
  </div>
</div>
`;

exports[`<DownloadList/> render empty 1`] = `<div />`;

exports[`<DownloadList/> render sounds with \`progress\` < 1 1`] = `
<div
  className="download-list"
>
  <div
    className="download active"
  >
    <div
      className="no-image"
    />
    <span
      className="title"
    >
      thunder
    </span>
    <div
      className="progress"
    >
      <div
        className="progress-barberpole"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <div
        className="progress-text"
      >
        50
        %
      </div>
    </div>
  </div>
  <div
    className="download active"
  >
    <div
      className="no-image"
    />
    <span
      className="title"
    >
      thunder
    </span>
    <div
      className="progress"
    >
      <div
        className="progress-barberpole"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <div
        className="progress-text"
      >
        50
        %
      </div>
    </div>
  </div>
  <div
    className="download active"
  >
    <div
      className="no-image"
    />
    <span
      className="title"
    >
      thunder
    </span>
    <div
      className="progress"
    >
      <div
        className="progress-barberpole"
        style={
          Object {
            "width": "50%",
          }
        }
      />
      <div
        className="progress-text"
      >
        50
        %
      </div>
    </div>
  </div>
</div>
`;
